src/components/channelList/channelList.js

```diff
@@ -0,0 +1,94 @@
+import News from '../news/news';
+import Channel from '../channel/channel';
+import { promiseDelay, showError} from '../../common';
+import Sender from '../../api';
+import { channels } from '../../data';
+
+export default class ChannelList {
+
+    constructor(spinner, header) {
+        this.spinner = spinner;
+        this.header = header;
+        this.listBlock = document.querySelector('.list');
+    }
+
+    buildChannels() {
+       
+        const elements = [];
+        channels.forEach((value, key) => {
+            const channel = new Channel(value, key);
+            const channelBlock = channel.render();
+            this.listBlock.appendChild(channelBlock);
```

Comment:
It is better to append channel blocks to a fragment and add to the DOM by one operation after the loop, what do you think?

Reply:
I agree with you. I'm going to resolve this issue.